source/util/StringUtils.cpp

```diff
@@ -16,13 +16,17 @@ namespace Aws
         {
             namespace Util
             {
+                // maximum buffer size after which log messages will get truncated
+                static const int MAX_BUF_SIZE = 4096;
                 string vFormatMessage(const char *message, va_list args)
                 {
                     va_list copy;
                     va_copy(copy, args);
-                    int formattedMsgSize = vsnprintf(nullptr, 0, message, copy) + 1;
+                    int formattedMsgSize = vsnprintf(nullptr, 0, message, copy);
                     va_end(copy);
 
+                    formattedMsgSize = std::min(formattedMsgSize, MAX_BUF_SIZE - 1) + 1;
```

Comment:
Since the log message bigger than `4096` will be dropped, we should document it.

Reply:
Do you think this is a reasonable size? Upon device client startup we log the entire config file, which is why I increased the limit. It is a little bit of an arbitrary decision.

Reply:
We cap the config file at 5k. Could align this limit to that constant which is just a magic number (linked below).
https://github.com/awslabs/aws-iot-device-client/blob/140913a61b35f1fed53506bf94fe96425f79f33c/source/config/Config.cpp#L1616

Reply:
yes

Reply:
@RogerZhongAWS  Can you make this a public constant visible and shared by Config.cpp and StringUtils.cpp?

Reply:
@RogerZhongAWS  Can you make this a public constant visible and shared by Config.cpp and StringUtils.cpp?